### PR TITLE
[Enhancement] Disable create materialized view for external table which created by resource

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

#11116

## Problem Summary(Required) ：

 Disable create materialized view for external table which created by resource

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|11|13|84.62%|[122, 191]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 1]
```diff
@@ -103,6 +106,22 @@ private boolean isSupportBasedOnTable(Table table) {
...
+            if (table instanceof OlapTable) {
+                return false;
+            } else if (table instanceof HiveTable || table instanceof HudiTable) {
+                HiveMetaStoreTable hiveMetaStoreTable = (HiveMetaStoreTable) table;
+                String catalogName = hiveMetaStoreTable.getCatalogName();
+                return Strings.isBlank(catalogName) || isResourceMappingCatalog(catalogName);
+            } else if (table instanceof IcebergTable) {
+                IcebergTable icebergTable = (IcebergTable) table;
+                String catalogName = icebergTable.getCatalog();
+                return Strings.isBlank(catalogName) || isResourceMappingCatalog(catalogName);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:119]: iceberg don't have resource mapping catalog.

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:119]: so I add strings.isblank

**Review by stephen-shelby — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|11|13|84.62%|[120, 189]|

**Review by stephen-shelby — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Youngwb** [comment]
run pr-title-check

**Comment by Youngwb** [comment]
run pr-title-check

**Comment by Youngwb** [comment]
@Mergify backport branch-2.5

**Pull request merged**